JustSaying.Messaging/MessageHandling/ExactlyOnceHandler.cs

```diff
@@ -21,7 +23,7 @@ public ExactlyOnceHandler(IHandler<T> inner, IMessageLock messageLock, int timeO
         
         public bool Handle(T message)
         {
-            var lockKey = string.Format("{0}-{1}-{2}", _inner.GetType().FullName.ToLower(), typeof(T).Name.ToLower(), message.UniqueKey());
```

Comment:
_inner is not suitable for this because it is always FutureHanlder<T> at this point.